File: 167. Two Sum II - Input Array Is Sorted/167. Two Sum II - Input Array Is Sorted.c
#include <stdlib.h>

/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* twoSum(int* numbers, int numbersSize, int target, int* returnSize) {
    int left = 0;
    int right = numbersSize - 1;

    // Allocate space for the result (2 integers)
    int* result = (int*)malloc(2 * sizeof(int));
    if (!result) return NULL; // malloc fail guard (optional)

    while (left < right) {
        int sum = numbers[left] + numbers[right];

        if (sum == target) {
            // Return 1-based indices
            result[0] = left + 1;
            result[1] = right + 1;
            *returnSize = 2;
            return result;
        }
        else if (sum > target) {
            right--;
        }
        else {
            left++;
        }
    }

    // If no solution found (per problem constraints, this should never happen)
    *returnSize = 0;
    free(result);
    return NULL;
}

File: 167. Two Sum II - Input Array Is Sorted/167. Two Sum II - Input Array Is Sorted.js
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(numbers, target) {
    let l = 0
    let r = numbers.length -1

    while (l <= r){
        if (numbers[l] + numbers[r] === target) return [l+1, r+1]
        else if (numbers [l] + numbers[r] > target) r = r - 1
        else l = l + 1
    }

    return
};